### ISSUE TITLE
System.Memory tests getting killed

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#20722]
I'm seeing this periodically breaking our full runs.

This guy is getting killed after only 10 seconds so I assume it's not timed out. @MattGal thoughts?
Unfortunately whatever is doing this doesn't lead to a dump. @shiftylogic @ahsonkhan 

```
2017-06-05 15:32:50,166: INFO: proc(54): run_and_log_output: Output: Copyright (C) 2014 Outercurve Foundation.
2017-06-05 15:32:50,167: INFO: proc(54): run_and_log_output: Output: 
2017-06-05 15:32:50,240: INFO: proc(54): run_and_log_output: Output: Discovering: System.Memory.Tests
2017-06-05 15:32:50,447: INFO: proc(54): run_and_log_output: Output: Discovered:  System.Memory.Tests
2017-06-05 15:32:50,606: INFO: proc(54): run_and_log_output: Output: Starting:    System.Memory.Tests
2017-06-05 15:33:01,766: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/f156f7a3-3468-4286-8536-e7b23f699283/Work/2433b65e-0236-4884-855d-3ffd16f67727/Unzip/RunTests.sh: line 89:  2205 Killed                  $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Memory.Tests.dll -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=failing
2017-06-05 15:33:01,777: INFO: proc(54): run_and_log_output: Output: command exited with ExitCode: 137
2017-06-05 15:33:01,777: INFO: proc(54): run_and_log_output: Output: command failed, trying to collect dumps
2017-06-05 15:33:01,826: INFO: proc(54): run_and_log_output: Output: corefile:
2017-06-05 15:33:01,826: INFO: proc(54): run_and_log_output: Output: no coredump file was found
2017-06-05 15:33:01,827: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/f156f7a3-3468-4286-8536-e7b23f699283/Work/2433b65e-0236-4884-855d-3ffd16f67727/Unzip
2017-06-05 15:33:01,828: INFO: proc(54): run_and_log_output: Output: Finished running tests. End time=15:33:01. Return value was 137
2017-06-05 15:33:01,834: INFO: proc(58): run_and_log_output: Exit Code: 137
2017-06-05 15:33:01,854: ERROR: scriptrunner(82): _main: Error: No exception thrown, but XUnit results not created
2017-06-05 15:33:01,856: ERROR: helix_test_execution(83): report_error: Error running xunit None
```
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170605.04/workItem/System.Memory.Tests/wilogs

**Comment by danmosemsft** [comment]
@mellinoe because i'd love for dumpling to be able to get a dump for killed/hung tests too.

**Comment by MattGal** [comment]
If you look at the Ubuntu 17.04 logs for this run you certainly see the Dumpling stuff get logged:

```
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: source Dumpling.sh
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: export XUNIT_PERFORMANCE_MIN_ITERATION=1
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: export XUNIT_PERFORMANCE_MAX_ITERATION=1
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: chmod +x /home/helixbot/dotnetbuild/work/1de319e6-0f63-4e7a-b80a-08c149547791/Payload/dotnet
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/1de319e6-0f63-4e7a-b80a-08c149547791/Payload/dotnet xunit.console.netcore.exe System.Memory.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
2017-06-05 15:35:03,325: INFO: proc(54): run_and_log_output: Output: CollectDumps 0 System.Memory.Tests
```

So if someone who works on this would like to help me investigate I'm  happy to but I think simply adding more logging to dumpling.sh might help.  From chatting with @mellinoe and looking at the dumpling site, we're definitely seeing some dumps from helixbot, it may simply not work on every distro that exists.

**Comment by ahsonkhan** [comment]
@shiftylogic, could this be related to the OOM killer given the error code?
https://stackoverflow.com/questions/30944355/bash-error-code-137-vs-1-when-out-of-memory

**Comment by mellinoe** [comment]
I'm about to merge a buildtools update which will pull in all of those Dumpling integration improvements I made last Friday. At the very least, we should get some better logging as to whether it was able to find a crash dump, what folder it looked it, what the exit code was, etc. Going forward we can add more logging in a centralized place (DumplingHelper.py) instead of two separate scripts.

**Comment by mellinoe** [comment]
That is all contingent on the process actually producing a core dump, though. Even the old script should have been able to find it if it was in the expected location. This should mean that there is no core file next to the application:

> Output: no coredump file was found

**Comment by shiftylogic** [comment]
I'm guessing it is the large memory Span::Clear test. It should probably be disabled on Linux. The deferred memory allocation + OOM killer is likely the cause of this.

**Comment by danmosemsft** [comment]
@shiftylogic could you please put up a PR to do this? if you disable it temporarily, we should leave this active with the disabled-test label.
@mellinoe yes I know there's no dump .. I'm just thinking aloud that ideally the infra would take one before it terminates the test process.

**Comment by shiftylogic** [comment]
@danmosemsft I doubt the infrastructure is what is killing this test. This appears to be the built-in Linux OOM killer that is taking the process down. At the point of the process getting killed, the test infrastructure likely just sees this as the test terminating. Depending on how the machine is configured, you might be able to see more information (re: OOM killer, what process, etc) in the syslog.

I'll submit a PR to disable the test on Linux. The only way we will be able to enable it (and other large memory tests) is to disable the memory overcommit in the Linux test machines. Though, I'm not sure how wise that would be. If these are short-lived machines that are recycled / rebooted after each test run, then that might not be a bad idea. If they are long-lived, this could lead to other problems.

**Comment by danmosemsft** [comment]
@shiftylogic makes sense. Re memory overcommit- see https://github.com/dotnet/corefx/issues/19923

**Comment by danmosemsft** [comment]
We will need a PR porting to 2.0 also as it happened there yesterday. Note I need to ask shiproom before hitting merge. https://mc.dot.net/#/product/netcore/200/source/official~2Fcorefx~2Frelease~2F2.0.0~2F/type/test~2Ffunctional~2Fcli~2F/build/20170605.01/workItem/System.Memory.Tests

**Comment by danmosemsft** [comment]
OK done.
